### PR TITLE
feat: add alpha.nvim integration

### PR DESCRIPTION
**Pull request by kylo252**
# Description

Add support for using https://github.com/goolord/alpha-nvim

Fixes performance issues with `dashboard.nvim`, as well provide the opportunity to use the `startify` mode.

## How Has This Been Tested?

```lua
lvim.builtin.alpha.mode = "dashbaord" -- or "startify"
```

### CODE REVIEW

**Comment by kylo252** [comment]
I haven't made any changes to the `startify` mostly because it was already a lot of work to fix `dashboard` 😅 

Here's a demo
40LRT5WOWZ

File: lua/lvim/core/alpha/startify.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
+local M = {}
+
+function M.config()
+  local startify = nil
+
+  return startify
```
Comment (abzcoding) [inline lua/lvim/core/alpha/startify.lua:6]: eh, what is going on here 😅 
`return nil` or what?

File: lua/lvim/plugins.lua [hunk 2]
```diff
@@ -177,6 +177,15 @@ return {
...
     disable = not lvim.builtin.dashboard.active,
   },
+  -- alpha
+  {
+    "goolord/alpha-nvim",
+    event = "BufWinEnter",
+    config = function()
+      require("lvim.core.alpha").setup()
+    end,
+    -- disable = not lvim.builtin.alpha.active,
```
Comment (abzcoding) [inline lua/lvim/plugins.lua:187]: i think we can still have
```lua
disable = not lvim.builtin.dashboard.active
```
or sth here

File: lua/lvim/core/alpha/startify.lua (hunk 1 above)
Comment (kylo252) [reply lua/lvim/core/alpha/startify.lua]: I wanted to add the integration, but I got tired yesterday 😄

File: lua/lvim/plugins.lua (hunk 2 above)
Comment (kylo252) [reply lua/lvim/plugins.lua:187]: Yeah I just need to make sure it's initialized before calling it.

**Comment by kylo252** [comment]
Updated demo

https://user-images.githubusercontent.com/59826753/140548247-805de3d1-695f-4f0e-b038-7d1f2e607e2d.mp4

**Comment by xeluxee** [comment]
Thanks for your work, but I think a user should be able to customize buttons (or at least change the keybinding shown on the right of the entries, because I've changed some of them in my `config.lua`)

**Comment by kylo252** [comment]
> Thanks for your work, but I think a user should be able to customize buttons (or at least change the keybinding shown on the right of the entries, because I've changed some of them in my `config.lua`)

They are customizable though, please check the last demo.
I even made sure it has a simpler API to use than the one shipped with alpha.

**Comment by xeluxee** [comment]
@kylo252 thanks I'd missed that
Just pointing out that `SPC p` keybinding to open recent projects isn't correct, because it opens packer options

**Comment by xeluxee** [comment]
What about displaying LunarVim version and the number of loaded packages similarly to dashboard?

**Comment by xeluxee** [comment]
Also how can I disable the autocmd to hide statusline in my `config.lua`?
I'd like statusline to be visible even in alpha buffer...

**Comment by xeluxee** [comment]
Any news about this?

**Comment by goolord** [comment]
just dropping this here https://github.com/goolord/alpha-nvim/commit/7ceb62ed560d7989585732728240db3ebffc5692 there is now an option to opt into using `noautocmd` when setting local options. 

this is apparently considerably faster for some people depending on how many FileType autocmds it fires but will prevent the alpha FileType autocmd from firing (thanks to the author of mini.nvim for discovering this lol https://github.com/goolord/alpha-nvim/issues/49)

**Comment by xeluxee** [comment]
@kylo252 what about [startup](https://github.com/startup-nvim/startup.nvim)?

**Comment by abzcoding** [comment]
Screen Shot 2022-02-12 at 5 19 20 PM

@kylo252 is this part intentional?
Screen Shot 2022-02-12 at 5 19 40 PM

**Comment by abzcoding** [comment]
Screen Shot 2022-02-12 at 8 24 59 PM

**Comment by abzcoding** [comment]
> Can you try calling that function directly and see what the return value is?

it was just a screenshot from GitHub actions. It should be fixed now

**Comment by kylo252** [comment]
> can we somehow remove the bottom right part

This one might be a bit more involved until https://github.com/goolord/alpha-nvim/pull/66 is merged, because I thought I fixed it here:
https://github.com/LunarVim/LunarVim/blob/a245484dfdc7695a0c89ef1c2482b4e3b7214905/lua/lvim/core/autocmds.lua#L56-L68

> the `support-alpha-nvim` before git commit?

I've done some refactoring to simplify this and make it more modular, feel free to change it:
https://github.com/LunarVim/LunarVim/blob/a245484dfdc7695a0c89ef1c2482b4e3b7214905/lua/lvim/core/alpha/dashboard.lua#L41-L46

**Comment by kylo252** [comment]
@abzcoding, any idea how to deal with this lualine error? https://github.com/LunarVim/LunarVim/runs/5611493978?check_suite_focus=true#step:4:26

**Comment by abzcoding** [comment]
> @abzcoding, any idea how to deal with this lualine error? https://github.com/LunarVim/LunarVim/runs/5611493978?check_suite_focus=true#step:4:26

well you found the solution 😅

**Review by abzcoding — APPROVED** [review]
LGTM 🚀

**Comment by kylo252** [comment]
> > @abzcoding, any idea how to deal with this lualine error? [LunarVim/LunarVim/runs/5611493978?check_suite_focus=true#step:4:26](https://github.com/LunarVim/LunarVim/runs/5611493978?check_suite_focus=true#step:4:26)
> 
> well you found the solution 😅

I actually don't like it my own solution... 😢  
I forgot that we might want to disable onedarker altogether. how's lualine finding that file?

**Comment by abzcoding** [comment]
> > > @abzcoding, any idea how to deal with this lualine error? [LunarVim/LunarVim/runs/5611493978?check_suite_focus=true#step:4:26](https://github.com/LunarVim/LunarVim/runs/5611493978?check_suite_focus=true#step:4:26)
> > 
> > 
> > well you found the solution 😅
> 
> I actually don't like my own solution... 😢 I forgot that we might want to disable onedarker altogether. how's lualine finding that file?

using this folder?
Screen Shot 2022-03-19 at 4 43 48 PM

**Review by abzcoding — APPROVED** [review]
👍

**Comment by kylo252** [comment]
still not quite sure what the issue was with lualine, but it could've been that `packer_on_complete` hook which I've disabled for now when in headless so hopefully no more issues!

**Review by abzcoding — APPROVED** [review]
LGTM 🚀

**Comment by m-rcussilva** [comment]
Hello guys! I'm completely lost. How can I fix this error or find the line 71? The init.lua file that I'm does not have the line 71.

Screen Shot 2022-03-28 at 11 04 25

Screen Shot 2022-03-28 at 11 05 20

**Comment by abzcoding** [comment]
@m-rcussilva do a `:PackerSync` , also as mentioned in the screenshot
`lvim.builtin.dashboard` has been replaced with `lvim.builtin.alpha`, please fix it inside `~/.config/lvim/config.lua`

**Comment by m-rcussilva** [comment]
Everything works fine now. Thank you @abzcoding !

**Pull request merged**